flood_adapt/object_model/io/unit_system.py

```diff
@@ -0,0 +1,360 @@
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+
+        return math.isclose(
+            self.value, other.convert(self.units), rel_tol=1e-2
+        )  # 1% relative tolerance for equality. So 1.0 == 1.01 evaluates to True
+
+    def __lt__(self, other):
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+        return self.value < other.convert(self.units)
+
+    def __gt__(self, other):
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+        return self.value > other.convert(self.units)
+
+    def __ge__(self, other):
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+        return (self > other) or (self == other)
+
+    def __le__(self, other):
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+        return (self < other) or (self == other)
+
+    def __ne__(self, other):
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                f"Cannot compare self: {type(self).__name__} to other: {type(other).__name__}"
+            )
+        return not (self == other)
+
+    def __mul__(self, other):
+        if isinstance(other, int) or isinstance(other, float):
+            return type(self)(self.value / other, self.units)
+        else:
+            raise TypeError(
+                f"Cannot multiply self: {type(self).__name__} with other: {type(other).__name__}. Only int and float are allowed."
+            )
+
+    def __div__(self, other):
+        if isinstance(other, int) or isinstance(other, float):
+            return type(self)(self.value / other, self.units)
+        elif isinstance(other, type(self)):
+            return self.value / other.convert(self.units)
+        else:
+            raise TypeError(
+                f"Cannot divide self: {type(self).__name__} with other: {type(other).__name__}. Only {type(self).__name__}, int and float are allowed."
+            )
+
+    def __truediv__(self, other):
+        if isinstance(other, int) or isinstance(other, float):
+            return type(self)(value=self.value / other, units=self.units)
+        elif isinstance(other, type(self)):
+            return self.value / other.convert(self.units)
+        else:
+            raise TypeError(
+                f"Cannot divide self: {type(self).__name__} with other: {type(other).__name__}. Only {type(self).__name__}, int and float are allowed."
+            )
```

Comment:
the units are gonna be wrong if you do this. if I multiply two lengths I'll get an area, not another length